packages/mobile/src/screens/chat-screen/ChatTextInput.tsx

```diff
@@ -60,63 +85,192 @@ export const ChatTextInput = ({
+      dispatch(sendMessage({ chatId, message: restoreLinks(inputMessage) }))
       setInputMessage('')
       onMessageSent()
+      clearLinks()
     }
-  }, [inputMessage, chatId, dispatch, onMessageSent])
+  }, [inputMessage, chatId, dispatch, onMessageSent, restoreLinks, clearLinks])
 
   const renderIcon = () => (
-    <Pressable
+    <TouchableOpacity
       onPress={handleSubmit}
       hitSlop={spacing(2)}
-      style={({ pressed }) => [
-        styles.iconCircle,
-        {
-          backgroundColor: hasLength
-            ? pressed
-              ? primaryDark2
-              : primary
-            : // Setting opacity style affects both the background and icon
-              // on Android, this workaround keeps the icon always white.
-              convertHexToRGBA(primary, 50)
-        }
-      ]}
+      style={styles.submit}
     >
       <IconSend
         width={styles.icon.width}
         height={styles.icon.height}
-        fill={styles.icon.fill}
+        fill={hasLength ? primary : neutralLight7}
       />
-    </Pressable>
+    </TouchableOpacity>
   )
 
+  const renderChatDisplay = (value: string) => {
+    const matches = getMatches(value)
+    if (!matches) {
+      const text = splitOnNewline(value)
+      return text.map((t, i) => <Text key={i}>{`${t}\n`}</Text>)
+    }
+    const parts: JSX.Element[] = []
+    let lastIndex = 0
+    for (const match of matches) {
+      const { index } = match
+      if (index === undefined) {
+        continue
+      }
+
+      if (index > lastIndex) {
+        // Add text before the match
+        const text = splitOnNewline(value.slice(lastIndex, index))
+        parts.push(
+          ...text.map((t, i) => (
+            <Text key={`${lastIndex}${i}`}>{`${t}\n`}</Text>
+          ))
+        )
+      }
+      // Add the matched word with accent color
+      parts.push(
+        <Text color='secondary' key={index}>
+          {match[0]}
+        </Text>
+      )
+      // Update lastIndex to the end of the current match
+      lastIndex = index + match[0].length
+    }
+
+    // Add remaining text after the last match
+    if (lastIndex < value.length) {
+      const text = splitOnNewline(value.slice(lastIndex))
+      parts.push(
+        ...text.map((t, i) => <Text key={`${lastIndex}${i}`}>{`${t}\n`}</Text>)
+      )
+    }
+
+    return parts
+  }
+
   return (
-    <TextInput
-      placeholder={messages.startNewMessage}
-      Icon={renderIcon}
-      styles={{
-        root: styles.composeTextContainer,
-        input: [
-          styles.composeTextInput,
-          Platform.OS === 'ios' ? { paddingBottom: spacing(1.5) } : null,
-          { maxHeight: hasCurrentlyPlayingTrack ? spacing(70) : spacing(80) }
-        ]
-      }}
-      onChangeText={setInputMessage}
-      inputAccessoryViewID='none'
-      multiline
-      value={inputMessage}
-      maxLength={10000}
-      autoCorrect
-    />
+    <Flex>
+      {trackId ? (
```

Comment:
just double-checking that we ensure `trackId` and `collectionId` can't both be set at the same time in the common hook

Reply:
they can be, but that's ok! it will show just a track in that case